compiler/x/amd64/codegen/OMRMemoryReference.cpp

```diff
@@ -171,27 +164,24 @@ void OMR::X86::AMD64::MemoryReference::finishInitialization(
       //
       mightNeedAddressRegister = false;
       }
-   else if (comp->getOption(TR_EnableHCR) && sr.getSymbol() && sr.getSymbol()->isClassObject())
+   else if (sr.getSymbol() != NULL && sr.isUnresolved())
       {
-      // Can't do any displacement-based tests because the displacement can change
+      // Once resolved, the address could be anything, so be conservative.
       //
       mightNeedAddressRegister = true;
       }
-   else if (self()->getBaseRegister() == NULL)
```

Comment:
it isn't clear to me why this case was eliminated? same for the getIndexRegister case?

Reply:
The information whether the base or index register is NULL is not related to whether address register may be needed. Therefore querying if either is NULL is useless.